src/controllers/searchController.ts

```diff
@@ -0,0 +1,33 @@
+import Product from '../models/Product';
+import {Request, Response, NextFunction} from 'express';
+
+async function searchProduct(req: Request, res: Response, next: NextFunction) {
+  try {
+    const query = cleanInput(req.params.query);
+    const product = await Product.sequelize?.query(
+      `
+        SELECT *
+        FROM "Products"
+        WHERE _search @@ to_tsquery('english', '${query}');
+      `,
+      {
+        model: Product,
+        replacements: {query: query},
+      }
+    );
+    if (product === null) {
```

Comment:
It is not exactly an error if search results are empty. Could return empty array with 200 status code.